### PR TITLE
Fix 'undefined reference to __data_start' linker error (Android/aarch64)

### PR DESCRIPTION
**Pull request by Over17**
This fix is needed to make embed-bitcode option work with Android NDK r23b.

Before r23b, we used bfd linker for Arm64 which apparently defined this symbol.
With r23b, we switch to lld and the symbol is no longer there, hence the fix is needed.

Confirmed to work locally.

### CODE REVIEW

**Review by joshpeterson — APPROVED** [review]

**Comment by joshpeterson** [comment]
Let's wait for @joncham to approve this as well before merging.

File: include/private/gcconfig.h
Comment (joncham) [inline include/private/gcconfig.h:2272]: i am not sure we want to do this. 
1. This is to enable the GC to scan native image for pointers. We should never need/enabled this in il2cpp or mono.
2. I will look, but I recall this having issues as it causes GC to scan a large hunk of memory unnecessarily.

Comment (joncham) [reply include/private/gcconfig.h:2272]: I am thinking of this old commit: https://github.com/Unity-Technologies/bdwgc/commit/5e73ff18487e2982f01dfc7913a62b77af7bc875

We need to make sure after any changes, we still don't push/scan the data segment of the image.

Comment (joshpeterson) [reply include/private/gcconfig.h:2272]: Wow, I remember debugging that now. That was a bad one. How did you ever come up with that?

Comment (Over17) [reply include/private/gcconfig.h:2272]: Okay if it has that much effect (scanning memory on startup, causing and catching a segfault), then we'd better refrain.

If `__data_start` is not needed/unused, can we have it ifdef'd or something?

Or what shall we do then?

The linker error (missing symbol) pops up only if `-fembed-bitcode` option is used.

**Comment by Over17** [comment]
Superseded by #71 .

**Pull request closed without merging**
